base/complex.jl

```diff
@@ -424,7 +424,8 @@ function log2(z::Complex)
     a/log(oftype(real(a),2))
 end
 
-function exp(z::Complex)
+exp(z::Complex) = exp(float(z))
```

Comment:
I think you need to rebase https://github.com/JuliaLang/julia/blob/fe29cb590d85df1d2c10ccc3c929f38418121f37/test/complex.jl#L945